src/mbgl/style/source_impl.cpp

```diff
@@ -199,6 +199,10 @@ std::unordered_map<std::string, std::vector<Feature>> Source::Impl::queryRendere
         return result;
     }
 
+    if (parameters.sourceIDs && parameters.sourceIDs->find(id) == parameters.sourceIDs->end()) {
```

Comment:
I'd prefer this check to be done in `Style::queryRenderedFeatures` - this way we can avoid the creation of `result` for sources not present in `sourceIDs`.